File: bolao_api/src/main/kotlin/com/capivara/bolao/model/Game.kt
package com.capivara.bolao.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*

class Game(@JsonProperty("id")
           var id: UUID?,
           @JsonProperty("championship_id")
           val championship: UUID,
           @JsonProperty("round_number")
           val roundNumber: Int,
           @JsonProperty("owner_team")
           val ownerTeam: GameTeam,
           @JsonProperty("visiting_team")
           val visitingTeam: GameTeam
)

File: bolao/lib/app/modules/championship/models/game.dart
import 'package:bolao/app/modules/championship/models/team.dart';
import 'package:bolao/app/modules/championship/models/team_base.dart';
import 'package:mobx/mobx.dart';

part 'game.g.dart';

class Game = _GameBase with _$Game;

abstract class _GameBase with Store {
  @observable
  Team ownerTeam;

  @observable
  Team visitingTeam;

  String championshipID;

  int roundNumber;

  _GameBase(TeamBase ownerTeam, TeamBase visitingTeam, this.championshipID, this.roundNumber) {
    this.ownerTeam = Team(ownerTeam.name, ownerTeam.imageURL, 0);
    this.visitingTeam = Team(visitingTeam.name, visitingTeam.imageURL, 0);
  }

  _GameBase.withScore(Team ownerTeam, Team visitingTeam, this.championshipID, this.roundNumber) {
    this.ownerTeam = ownerTeam;
    this.visitingTeam = visitingTeam;
  }

  String getScoreboard() {
    return "${ownerTeam.name}\t${ownerTeam.score}\t${visitingTeam.score}\t${visitingTeam.name}";
  }
}
